Assets/Scripts/UI/Components/ATM/Timeline_Regulator.cs

```diff
@@ -0,0 +1,237 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+using UnityEngine.EventSystems;
+
+namespace Netherlands3D.Twin
+{
+    public class Timeline_Regulator : MonoBehaviour
+    {
+        public GameObject yearPrefab; // Prefab for the year instance  
+        public RectTransform content; // Content RectTransform of the ScrollView  
+        public TMP_InputField yearInputField; // Input field for the year  
+        public TMP_InputField centerYearText; // Text object to display the centered year  
+        public Slider speedSlider; // Slider to control year modification  
+        int minYear = 1625;
+        int maxYear = 2024;
+        public TMP_InputField startYearInput; // Reference to the start year input field  
+        public TMP_InputField endYearInput; // Reference to the end year input field  
+
+        private List<RectTransform> yearInstances = new List<RectTransform>();
+        public int currentYear;
+
+        private float lerpSpeed = 10f; // Speed of the lerp transition  
+        private Vector2 targetPosition; // Target position for the content  
+        private Coroutine yearUpdateCoroutine; // Reference to the coroutine
+
+        // Define original, highlighted and neighbor height values  
+        private float defaultHeight = 3; // Original height of YearPrefab  
+        private float highlightedHeight = 12f; // Height for the current year  
+        private float neighborHeight = 8; // Height for adjacent years
+
+        private Color highlightedColor = new Color(0f, 0.274f, 0.6f); // HEX #004699  
+        private Color defaultColor = new Color(0.8f, 0.847f, 0.894f); // HEX #CCD7E4
+
+        // List of years with corresponding events  
+        public List<int> eventYears = new List<int>() { 1625, 1700, 1800, 1878, 1900, 1930, 1949, 1999, 1989, 1949, 2000, 2024 }; // Example years
+
+        void Start()
+        {
+            // Set the initial years to the default values  
+            minYear = int.Parse(startYearInput.text);
+            maxYear = int.Parse(endYearInput.text);
+
+            InitializeTimeline();
+
+            // Call ScrollToYear after a brief delay to ensure UI is fully initialized  
+            StartCoroutine(ScrollToMaxYearAfterDelay());
+
+            // Add listeners to the input fields to detect changes  
+            startYearInput.onEndEdit.AddListener(delegate { UpdateTimeline(); });
```

Comment:
je hoeft hier geen delegate van te maken, het kan ook zo:
`startYearInput.onEndEdit.AddListener(UpdateTimeline);`

Reply:
ah het was omdat die endEdit een string returnt, maar omdat je allebei de input field texts nodig hebt is dat niet nodig, dus ik laat het wel zo